### PR TITLE
nodejs: pin to 12.x by default

### PR DESCRIPTION
**Pull request by disassembler**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by disassembler** [comment]
This is not going in release 20.03. It will be part of 20.09.

**Comment by marsam** [comment]
I think we should also make `nodePackages` to use nodejs-12_x

File: pkgs/development/node-packages/composition-v13.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 {pkgs ? import <nixpkgs> {
     inherit system;
-  }, system ? builtins.currentSystem, nodejs ? pkgs."nodejs-13_x"}:
+  }, system ? builtins.currentSystem, nodejs ? pkgs."nodejs-8_x"}:
```
Comment (raboof) [inline pkgs/development/node-packages/composition-v13.nix:5]: This seems wrong?

File: pkgs/development/node-packages/node-packages-v10.nix [hunk 2]
```diff
@@ -8941,13 +8941,13 @@ let
         sha512 = "bsTwuIg/BZZK/vreVTYYbSWoe2F+71P7K5QGEX+pT250DZbfU1MQ5prOKpPR+LL6uWKK3KMwMCAS74QB3Um1uw==";
       };
     };
-    "caniuse-lite-1.0.30001025" = {
+    "caniuse-lite-1.0.30001027" = {
```
Comment (raboof) [inline pkgs/development/node-packages/node-packages-v10.nix:8944]: Are these changes relevant here?

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -4677,7 +4677,7 @@ in
 
   nixnote2 = libsForQt5.callPackage ../applications/misc/nixnote2 { };
 
-  nodejs = hiPrio nodejs-10_x;
+  nodejs = hiPrio nodejs-12_x;
```
Comment (raboof) [inline pkgs/top-level/all-packages.nix:4680]: :+1: I would like this, too. Perhaps also update `nodejs-slim`?

**Review by raboof — COMMENTED** [review]
Some comments, and it seems this PR has a merge conflict so needs a rebase.

**Comment by prusnak** [comment]
Seems this is superseded by https://github.com/NixOS/nixpkgs/pull/85764

**Pull request closed without merging**